azure/services/managedclusters/spec_test.go

```diff
@@ -379,6 +380,48 @@ func TestParameters(t *testing.T) {
 				g.Expect(result).To(BeNil())
 			},
 		},
+		{
+			name: "setting networkPluginMode from nil to \"overlay\" will update",
```

Comment:
Is this a valid use case? (i.e., does AKS allow us to change this value to an existing cluster?)

Reply:
Yes: https://learn.microsoft.com/en-us/azure/aks/azure-cni-overlay#upgrade-an-existing-cluster-to-cni-overlay

Reply:
Are there any CAPZ apiserver webhooks we can add to protect these potential failures?

- Doesn't use the dynamic pod IP allocation feature.
- Doesn't have network policies enabled.
- Doesn't use any Windows node pools with docker as the container runtime.

(i.e., does CAPZ support the above features that don't work w/ Azure CNI overlay)

Reply:
Where would we check for each of those things? If they're defined in CAPZ objects then we could check those. I'm not sure if we have plumbing set up in the webhooks to do Azure API calls though.

Reply:
Right, we would only do this effectively if there was a CAPZ representation that corrrelated to the violating AKS features. If we haven't yet implemented support for the above then we would probably not want to mess around with pre-validating against the AKS API mid-flight (we should defer to AKS on that and improve the UX by actually implementing CAPZ wrapping of those features, IMO).

We do seem to support NetworkPolicy in CAPZ, so maybe we could add a webhook to enforce that we don't ship that config w/ CNI overlay.

Reply:
I can add a check for network policy. I don't think we can (easily) add anything for those other items though.